controllers/helpers.go

```diff
@@ -245,6 +251,19 @@ func systemAssignedIdentityCloudProviderConfig(d azure.ClusterScoper) (cpConfig
 	return controlPlaneConfig, workerConfig
 }
 
+func workloadIdentityCloudProviderConfig(d azure.ClusterScoper) (cpConfig *CloudProviderConfig, wkConfig *CloudProviderConfig) {
+	controlPlaneConfig, workerConfig := newCloudProviderConfig(d)
+	// secret is not needed in workload identity.
+	controlPlaneConfig.AadClientSecret = ""
+	controlPlaneConfig.UseFederatedWorkloadIdentityExtension = true
+	controlPlaneConfig.AADFederatedTokenFile = aadFederatedTokenFilePath
```

Comment:
For this to actually work with the workload cluster, the `--service-account-issuer-url` needs to be configured with an OIDC issuer URL that's https and publicly accessible serving the discovery document and jwks endpoint.

Reply:
The workload cluster that will be created will have the `--service-account-issuer-url` provided via the workload cluster template. 

Copying the yaml from the PR description 
```
  kubeadmConfigSpec:
    clusterConfiguration:
      apiServer:
        extraArgs:
          cloud-provider: external
          # This flag needs to be passed for the workload cluster to be able to use workload identity
          service-account-issuer: <your-service-issuer-url>
          # This flag needs to be passed for the workload cluster to be able to use workload identity
          service-account-key-file: /etc/kubernetes/pki/sa.pub
          # This flag needs to be passed for the workload cluster to be able to use workload identity
          service-account-signing-key-file: /etc/kubernetes/pki/sa.key
        timeoutForControlPlane: 20m
      controllerManager:
        extraArgs:
          allocate-node-cidrs: "false"
          cloud-provider: external
          cluster-name: azwi-cluster
          # This flag needs to be passed for the workload cluster to be able to use workload identity
          service-account-private-key-file: /etc/kubernetes/pki/sa.key
 ```